store/tikv/mock-tikv/cluster.go

```diff
@@ -186,22 +198,141 @@ func (c *Cluster) Split(regionID, newRegionID uint64, key []byte, peerIDs []uint
+}
+
+// SplitIndex evenly splits the data in index into count regions.
+// Only works for single store.
+func (c *Cluster) SplitIndex(mvccStore *MvccStore, tableID, indexID int64, count int) {
+	indexStart := tablecodec.EncodeTableIndexPrefix(tableID, indexID)
+	indexEnd := indexStart.PrefixNext()
+	c.splitRange(mvccStore, NewMvccKey(indexStart), NewMvccKey(indexEnd), count)
+}
+
+func (c *Cluster) splitRange(mvccStore *MvccStore, start, end MvccKey, count int) {
+	c.evacuateOldRegionRanges(start, end)
+	regionPairs := c.getEntriesGroupByRegions(mvccStore, start, end, count)
+	c.createNewRegions(regionPairs, start, end)
+}
+
+// getPairsGroupByRegions groups the key value pairs into splitted regions.
+func (c *Cluster) getEntriesGroupByRegions(mvccStore *MvccStore, start, end MvccKey, count int) [][]Pair {
+	startTS := uint64(math.MaxUint64)
+	limit := int(math.MaxInt32)
+	pairs := mvccStore.Scan(start.Raw(), end.Raw(), limit, startTS)
+	regionEntriesSlice := make([][]Pair, 0, count)
+	quotient := len(pairs) / count
+	remainder := len(pairs) % count
+	i := 0
+	for i < len(pairs) {
+		regionEntryCount := quotient
+		if remainder > 0 {
+			remainder--
+			regionEntryCount++
+		}
+		regionEntries := pairs[i : i+regionEntryCount]
+		regionEntriesSlice = append(regionEntriesSlice, regionEntries)
+		i += regionEntryCount
+	}
+	return regionEntriesSlice
+}
+
+func (c *Cluster) createNewRegions(regionPairs [][]Pair, start, end MvccKey) {
+	storeID, _ := c.firstStoreAndPeerID()
+	for i := range regionPairs {
+		peerID := c.allocID()
+		newRegion := newRegion(c.allocID(), []uint64{storeID}, []uint64{peerID}, peerID)
+		if i == 0 {
+			newRegion.updateKeyRange(start, NewMvccKey(regionPairs[i+1][0].Key))
+		} else if i == len(regionPairs)-1 {
+			newRegion.updateKeyRange(NewMvccKey(regionPairs[i][0].Key), end)
+		} else {
+			newRegion.updateKeyRange(NewMvccKey(regionPairs[i][0].Key), NewMvccKey(regionPairs[i+1][0].Key))
+		}
+		c.regions[newRegion.Meta.Id] = newRegion
+	}
+}
+
+// evacuateOldRegionRanges evacuate the range [start, end].
+// Old regions has intersection with [start, end) will be updated or deleted.
+func (c *Cluster) evacuateOldRegionRanges(start, end MvccKey) {
+	storeID, peerID := c.firstStoreAndPeerID()
+	oldRegions := c.getRegionsCoverRange(start, end)
+	for _, oldRegion := range oldRegions {
+		startCmp := bytes.Compare(oldRegion.Meta.StartKey, start)
+		endCmp := bytes.Compare(oldRegion.Meta.EndKey, end)
+		if len(oldRegion.Meta.EndKey) == 0 {
+			endCmp = 1
+		}
+		if startCmp >= 0 && endCmp <= 0 {
+			// The region is within table data, it will be replaced by new regions.
+			delete(c.regions, oldRegion.Meta.Id)
+		} else if startCmp < 0 && endCmp > 0 {
+			// a single Region covers table data, split into two regions that do not overlap table data.
+			oldEnd := oldRegion.Meta.EndKey
+			oldRegion.updateKeyRange(oldRegion.Meta.StartKey, start)
+
+			newRegion := newRegion(c.allocID(), []uint64{storeID}, []uint64{peerID}, peerID)
```

Comment:
Allocate a new peerID.
And remove the function of `firstStoreAndPeerID`. Use `c.stores[0]` to get storeID.